### PR TITLE
ELF geolocator result sorting fix

### PR DESCRIPTION
**Pull request by data-ux**
Ensure exact search result is first in list

### CODE REVIEW

File: service-search-nls/src/main/java/fi/nls/oskari/search/util/ScoredSearchHit.java [hunk 1]
```diff
@@ -3,14 +3,20 @@
 class ScoredSearchHit implements Comparable<ScoredSearchHit> {
     public final String text;
     public final int score;
+    public final Boolean isExact;
```
Comment (jampukka) [inline service-search-nls/src/main/java/fi/nls/oskari/search/util/ScoredSearchHit.java]: isExact could be primitive boolean. You can use Boolean.compare(boolean a, boolean b); for comparisons if that's the reason you chose the Object version.

File: service-search-nls/src/main/java/fi/nls/oskari/search/util/ScoredSearchHit.java [hunk 2]
```diff
@@ -3,14 +3,20 @@
...
         score = s;
+        isExact = e;
     }
 
     @Override
     public int compareTo(ScoredSearchHit other) {
-        return score - other.score;
+        int comparison = this.isExact.compareTo(other.isExact);
+        if (comparison == 0) {
+            return score - other.score;
```
Comment (jampukka) [inline service-search-nls/src/main/java/fi/nls/oskari/search/util/ScoredSearchHit.java]: Integer.compare(int a, int b); would handle edge cases better, most probably won't be a problem here, but for future reference.

File: service-search-nls/src/main/java/fi/nls/oskari/search/channel/ELFGeoLocatorSearchChannel.java [hunk 3]
```diff
@@ -448,12 +448,14 @@ public void calculateCommonFields(final SearchResultItem item) {
...
+                combiner.addHit(fuzzyHits.getJSONObject(i), false);
             }
 
-            JSONArray exactHits = jsonObject.getJSONArray("normal_search").getJSONObject(0).getJSONArray("options");
-            for (int i = 0; i < exactHits.length(); i++) {
-                combiner.addHit(exactHits.getJSONObject(i), 1000);
+            JSONArray normalHits = jsonObject.getJSONArray("normal_search").getJSONObject(0).getJSONArray("options");
+            for (int i = 0; i < normalHits.length(); i++) {
+                JSONObject hit = normalHits.getJSONObject(i);
+                boolean isExact = searchString.trim().equalsIgnoreCase(hit.getString("text"));
```
Comment (jampukka) [inline service-search-nls/src/main/java/fi/nls/oskari/search/channel/ELFGeoLocatorSearchChannel.java]: Could move the trim() outside of the loop, however that might get optimized by the compiler anyway.

Comment (data-ux) [reply service-search-nls/src/main/java/fi/nls/oskari/search/channel/ELFGeoLocatorSearchChannel.java]: Yes, youre right, but I left this as is, beacause the performance impact is minimal with the amount of results we have here. I think it's more readable this way.

**Comment by jampukka** [comment]
Looks good to me

**Pull request merged**